### DISCUSSION TITLE
Unknown column 'email' in php artisan make:filament-user

### DISCUSSION THREAD

**Discussion by rzrusu** [filamentphp/filament#12521] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.57

### How can we help you?

```SQLSTATE[42S22]: Column not found: 1054 Unknown column 'email' in 'where clause' (Connection: mysql, SQL: select exists(select * from `account` where `email` = myemail@gmail.com) as `exists`) ``` 

Trying to set up Filament for my project, and this is the error I get when I try running `php artisan make:filament-user`. 
I have customized my project to use different column names for authentication, in this case _'email'_ is now _'AccountEmail_', and _'users'_ table is now _'account'_. It seems to use the right table, but it's not using the right column name for email. All other Auth features work with the custom names, it's just this giving me trouble. 

This is what my User model looks like: 
```
...
class User extends Authenticatable
{
    use HasFactory, Notifiable;

    //protected $connection = 'sprp';
    protected $table = 'account';

    protected $primaryKey = 'AccountID';

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'AccountName',
        'AccountPassword',
        'AccountEmail',
    ];

    /**
     * The attributes that should be hidden for serialization.
     *
     * @var array<int, string>
     */
    protected $hidden = [
        'AccountPassword',
        'remember_token',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'EmailVerifiedAt' => 'datetime',
            'AccountPassword' => 'hashed',
        ];
    }

    public function hasVerifiedEmail() {
        return is_null($this->EmailVerifiedAt);
    }

    public function getAuthPassword() {
        return $this->AccountPassword;
    }

    public function username()
    {
        return 'AccountEmail';
    }

    public function getEmailForPasswordReset()
    {
        return $this->AccountEmail;
    }
}

```

You can see I've overridden functions like username() and getEmailForPasswordReset() for email but it doesn't seem to have any effect. Any ideas would be highly appreciated. Thanks!

**Comment by leandrocfe** [comment, marked as answer]
By default, Filament uses the “email” column. If you decide to change it, make sure to update all relevant places, including the specific command you’re referring to.
